addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js

```diff
@@ -90,7 +91,13 @@ class LaunchProduct extends StepBase {
 
     // Read input params specified in the environment type configuration
     // The params may include variable expressions, resolve the expressions by using the resolveVars
+    // By doing this resolution, we might overwrite the dynamic, unique value defined above with a static name 
+    // that would not be unique between deployments of the same workspace configuration
     const resolvedInputParams = await this.resolveVarExpressions(envTypeConfig.params, resolvedVars);
+    // Additional layer to check the namespace is valid and unique. If not, make new namespace from
+    // static and get index of namespace param to change
+    const { namespaceParam, namespaceIndex } = this.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
+    resolvedInputParams[namespaceIndex].Value = namespaceParam;
```

Comment:
Instead of assigning the value of `namspace` to `resolvedVars` twice, we could do it just once right? Is there a reason why we're setting the value twice? Setting the value twice could lead to user error in the future as they might not realized that `namespace` is being rewritten soon after it was initially set on line 87.

For example, we could set it once like this.
https://github.com/awslabs/service-workbench-on-aws/pull/615#discussion_r680236315

Reply:
I was setting it twice because the original set provides us some way to check if it was changed during the resolveVarExpressions/envTypeConfigService.mustfind call. I thought it was a simple way to check if the param was static or not. Do you have another idea or a better way to do that?

Reply:
I think line 87 is needed actually, since in function [resolveVarExpressions](https://github.com/awslabs/service-workbench-on-aws/blob/3acd6d983db7d51f502a7a8e17793992be1234a7/addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js#L324), it'll be resolving a placeholder value `${namespace}` to what's set in line 87. If the value does not exist, lodash will throw error `namespace is not defined`.

Reply:
that makes sense. thank you for the explanation